apollo-router/src/plugins/telemetry/reload.rs

```diff
@@ -166,20 +166,18 @@ where
 {
     fn enabled(
         &self,
-        meta: &tracing::Metadata<'_>,
+        _meta: &tracing::Metadata<'_>,
         cx: &tracing_subscriber::layer::Context<'_, S>,
     ) -> bool {
         let current_span = cx.current_span();
 
-        //
-        !meta.is_span()
-            // this span is enabled if:
-            || current_span
-                .id()
-                // - there's a parent span and it was enabled
-                .map(|id| cx.span(id).is_some())
-                // - there's no parent span (it's the root), so we make the sampling decision
-                .unwrap_or_else(|| self.sample())
+        // this span is enabled if:
+        current_span
+            .id()
+            // - there's a parent span and it was enabled
+            .map(|id| cx.span(id).is_some())
+            // - there's no parent span (it's the root), so we make the sampling decision
+            .unwrap_or_else(|| self.sample())
```

Comment:
I don't really understand this filtering, but it feels like it should be:
```
meta.is_span() &&
current_span
            .id()
            // - there's a parent span and it was enabled
            .map(|id| cx.span(id).is_some())
            // - there's no parent span (it's the root), so we make the sampling decision
            .unwrap_or_else(|| self.sample())
```
I don't understand why we wouldn't include the metadata in the `enabled` decision since it would be a quick exit if it isn't a span.

Reply:
it's been done, thanks for the suggestion